src/steps/ecs/types.ts

```diff
@@ -0,0 +1,143 @@
+export interface Instance {
+  InstanceId: string;
+  InstanceName?: string;
+  InstanceType?: string;
+  RegionId?: string;
+  CreationTime?: string;
+  SerialNumber?: string;
+  Status?: string;
+  DeploymentSetId?: string;
+  KeyPairName?: string;
+  SaleCycle?: string;
+  SpotStrategy?: string;
+  DeviceAvailable?: boolean;
+  LocalStorageCapacity?: number;
+  Description?: string;
+  SpotDuration?: number;
+  InstanceNetworkType?: string;
+  OSNameEn?: string;
+  HpcClusterId?: string;
+  SpotPriceLimit?: number;
+  Memory?: number;
+  OSName?: string;
+  DeploymentSetGroupNo?: number;
+  ImageId?: string;
+  VlanId?: string;
+  ClusterId?: string;
+  GPUSpec?: string;
+  AutoReleaseTime?: string;
+  DeletionProtection?: boolean;
+  StoppedMode?: string;
+  GPUAmount?: number;
+  HostName?: string;
+  InternetMaxBandwidthOut?: number;
+  InternetMaxBandwidthIn?: number;
+  InstanceChargeType?: string;
+  IoOptimized?: boolean;
+  StartTime?: string;
+  Cpu?: number;
+  LocalStorageAmount?: number;
+  ExpiredTime?: string;
+  ResourceGroupId?: string;
+  InternetChargeType?: string;
+  ZoneId?: string;
+  Recyclable?: boolean;
+  ISP?: string;
+  CreditSpecification?: string;
+  InstanceTypeFamily?: string;
+  OSType?: string;
+  NetworkInterfaces?: {
+    NetworkInterface: NetworkInterface[];
+  };
+  OperationLocks?: {
+    LockReason: LockReason[];
+  };
+  Tags?: {
+    Tag: Tag[];
+  };
```

Comment:
Noticed that these should also probably be optional in this interface: 

```suggestion
  NetworkInterfaces?: {
    NetworkInterface?: NetworkInterface[];
  };
  OperationLocks?: {
    LockReason?: LockReason[];
  };
  Tags?: {
    Tag?: Tag[];
  };
```

Reply:
:+1: will update